### DISCUSSION TITLE
How can we join the review process of the document translation?

### DISCUSSION THREAD

**Discussion by AseiSugiyama** [tensorflow/docs-l10n#418] [Q&A]
Since I had been away from the translation community, I have some questions about the process of the docs-l10n community.

## Onboarding contents

I have read [README.md](https://github.com/tensorflow/docs-l10n/blob/master/README.md) and [TensorFlow docs contributor guide](https://www.tensorflow.org/community/contribute/docs#translations). Is there any other document to check before starting the document translation process?

## New translation process

We had the review process based on PRs. Reviewers are described on REVIEWERS like  [Japanese community's REVIEWERS](https://github.com/tensorflow/docs-l10n/blob/master/site/ja/REVIEWERS) and all PRs were required to be reviewed by them before deploying it to https://www.tensorflow.org/

How can we join the reviewing process? Is there any document about it?

I will introduce the new way to join the translation process to the Japanese translation community after I got some answers.

**Comment by AseiSugiyama** [comment]
@lamberta Could you give me any suggestions about the above questions?

**Comment by lamberta** [comment]
Hi @AseiSugiyama (sorry, I missed this thread)

We've integrated [GitLocalize](https://gitlocalize.com/tensorflow/docs-l10n) into our translation pipeline for vendors and the community to submit pull requests. You should be able to add reviews in the [Japanese section](https://gitlocalize.com/repo/4592/ja/site/en-snapshot).

> We had the review process based on PRs. Reviewers are described on REVIEWERS

This has changed a bit since relying on community reviewers is a bottleneck and we want to increase coverage. The Japanese community has done an awesome job, but there's just a lot of docs on tensorflow.org. We're hoping a hybrid model of vendors + community will give use the coverage and quality we need. Please do add reviews or submit PRs using GitLocalize, but we probably need to clarify what that community process looks like. cc: @ilyaspiridonov

**Reply by AseiSugiyama** [reply]
Agree with you, the vendor's work is amazing, both quality and quantity, and as a community, I guess we can contribute to the translation of the docs. To realize a hybrid model, it is useful to make the translation process clear. I would like to discuss the followings;

1. Which document is in the process of the vendor's translation?  Which one is ready to be translated by the community? How can we know it?
2. Can the community join the review? If yes, how can we do that?

I think it is very important to avoid reproducing the bottleneck. I hope to make the translation better by realizing the hybrid model.

**Comment by lamberta** [comment]
Hi @AseiSugiyama 

Please look through the GitLocalize UI for the [Japanese section](https://gitlocalize.com/repo/4592/ja/site/en-snapshot). To the right you'll see some stats including translation coverage. This will give you an idea of what sections need work. If you click on the [Review Requests](https://gitlocalize.com/repo/4592/ja/review) you'll see the open translations (not submitted as pull requests, yet). If you click on the entry you can comment on the translation.

**Reply by AseiSugiyama** [reply]
Thank you for your suggestion, I've succeed in making a [Review Request](https://gitlocalize.com/repo/4592/ja/review/26820). Thank you for your support!

**Comment by sfujiwara** [comment]
@lamberta I tried to translate a doc and create a review request, but `Create Review Request` button is disable.

![image](https://user-images.githubusercontent.com/2109535/100377647-f3c39780-3054-11eb-8732-d8507a6a7cf3.png)

I guess that is because we don't have a permission of `translator`. It can be set in Team configuration (the below is screen shot of my own repository for sandbox):

![image](https://user-images.githubusercontent.com/2109535/100377779-48671280-3055-11eb-8311-01be6b0307dc.png)

**Comment by sfujiwara** [comment]
Sorry, it seems that we can not create review request if the coverage is not 100%.
I fixed it and tried to create review request!
https://gitlocalize.com/repo/4592/ja/review/25915

**Comment by ilyaspiridonov** [comment]
@sfujiwara Hi, this is Ilya, and I represent GitLocalize. Correct, we currently need the file to be 100% translated before it can be sent as Pull Request. 
Should you have any questions related to working with GitLocalize -feel free to ping our support team via the chat widget on Gitlocalize.com or in our [Gitter](https://gitter.im/gitlocalize/Lobby)

**Reply by AseiSugiyama** [reply]
I and @sfujiwara were able to make two Review Request (https://gitlocalize.com/repo/4592/ja/review/25915 & https://gitlocalize.com/repo/4592/ja/review/26820 ). Thanks!

**Comment by AseiSugiyama** [comment]
@ilyaspiridonov @lamberta Sorry for multiposting (especially for @ilyaspiridonov ), but please let me know how we can contribute the translation of docs. We have created two Review Requests and they have not been reviewed. One of them has been outdated until waiting reviews.

AFAIK, Google and Alconost are going to provide the process to support community translations at es ( https://github.com/tensorflow/docs-l10n/discussions/449#discussioncomment-395348 ). What do you think about the Japanese language?

**Reply by ilyaspiridonov** [reply]
Hello @AseiSugiyama! First of all thanks for your support in adding and updating translations! 
We have worked out, together with @lamberta, a plan to continuously work on JA, ZH and KO translations in Q1-Q2 2021. As part of that, Alconost is now working on the next batch of translations. We were planning to review and send outstanding Review Requests from the community by end of Q1 as part of that process, but I will ask our team to see if they can prioritize this. Community translations will then be reviewed on an ongoing basis as they are added (presumably, once every couple weeks). This is becoming more of a workflow, please bear with us!

**Reply by sfujiwara** [reply]
@ilyaspiridonov Thanks. Which is better for our community to start translation or wait for Alconost's works?
I'm concerned about duplication of  community's translation and Alconost's one.

**Reply by lamberta** [reply]
Thanks, @AseiSugiyama and @sfujiwara, for your continued interest in this project. Now that we have the pipeline integrated we'll focus on the workflows and reviews.

For JA, KO, and ZH-CN, the community's role will shift from before—now we shouldn't have to worry about coverage since the vendors work on quarterly snapshots. Once we get the community review process established, the community would be best utilized to improve quality and fix [broken notebooks](https://github.com/tensorflow/docs-l10n/blob/master/site/ja/FAILURES.md).

As far as duplicating review work. everything is synced through the GitHub repo. But maybe we can toggle a flag somewhere as a signal that the job has started or a file is out for vendor translation? Once things settle we can probably tie these jobs to dates.

**Reply by AseiSugiyama** [reply]
Thank you for your reply @ilyaspiridonov and @lamberta . I think it is important to fix broken notebooks, help to improve quality, and other tiny fix we found (like language special missing link) and I am interested to contribute these things. I'll create very small pull request and waiting the day the workflow should be announced.

**Comment by CaothicPechan** [comment]
Hi, @ilyaspiridonov I'm willing to contribute to translating the Spanish documents, Is there any consideration I should take before doing this?
